Sources/TokamakDOM/Views/NavigationLink.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TokamakCore
+
+public typealias NavigationLink = TokamakCore.NavigationLink
+
+extension NavigationLink: ViewDeferredToRenderer {
+  public var deferredBody: AnyView {
+    let proxy = _NavigationLinkProxy(self)
+    return AnyView(
+      HTML("div", listeners: ["click": { _ in proxy.activate() }]) { proxy.label }
```

Comment:
Wouldn't we want to use a simple `a` here?

```suggestion
      HTML("a", ["href': "#"], listeners: ["click": { _ in proxy.activate() }]) { proxy.label }
```

Reply:
Definitely for a proof of concept but I wanted to have support for different things that can go inside NavigationLinks and don’t necessarily look like links (for example, `List` rows)

Reply:
Hm, how would an example with `List` rows look code-wise?

Reply:
like

```swift
List {
  NavigationLink("First Last", destination: ...)
}
```

for the list in the Contacts iOS app

Reply:
Right, but that still would make sense to render as a sequence of `<a>` tags, wouldn't it?

Reply:
At least when `Label` type is `Text`. But even when `Label` type is say `Image`, it would be an `<a>` tag wrapping `<img>` inside of it anyway.

Reply:
(At least in my opinion, but I'm happy consider counter-examples 🙂)

Reply:
Here’s what it does on macOS

<img width="751" alt="Screenshot_2020-07-02 14 57 59" src="https://user-images.githubusercontent.com/25517624/86398945-71d16d00-bc74-11ea-8e3b-a0e37ea991ae.png">

Maybe a link is a good default but an environment variable of some sort could be added in the future to allow further customization?

Reply:
Yeah, I think it could be controlled with something like `NavigationLinkStyle`, doesn't seem to be present in SwiftUI, but that's the name I'd give it 😄

Reply:
But even if we consider something like `NavigationLinkStyle`, probably not worth including it in this PR, implementing basic defaults is a good enough approach for now I think. As soon as we see more people using it and giving feedback, we'll understand what they want and how we should go forward with customizations like this.

Reply:
I did some additional testing and it seems like if you wrap the `NavigationLink` in modifiers or layout views, it retains the unstyled appearance but if you put it inside of an actual UI element (like a toggle) it turns back into a button.

Reply:
Wouldn't `NavigationLink { Button { Text("Foo") } }` rendered as `<a href="#"><button>foo</button></a>` work essentially the same? 🤔

Or do you mean `Button { NavigationLink { Text("Foo") } }` by "if you put it inside of an actual UI element"?

Reply:
On the Mac, I did `Toggle { NavigationLink("Foo") }` and it worked strangely. I’m going to just try to push the link version when I can get it running in browser again.